Sources/App/Commands/Analyzer.swift

```diff
@@ -57,18 +81,27 @@ func analyze(application: Application, limit: Int) throws -> EventLoopFuture<Voi
 
     // FIXME: sas 2020-05-04: Workaround for partial flush described here:
     // https://discordapp.com/channels/431917998102675485/444249946808647699/706796431540748372
+    // FIXME: this breaks exposing this via the API:
+    // Precondition failed: BUG DETECTED: wait() must not be called when on an EventLoop.
     let fulfilledUpdates = try versionUpdates.wait()
     let setStatus = fulfilledUpdates.map { (pkg, updates) -> EventLoopFuture<[Void]> in
-        EventLoopFuture.whenAllComplete(updates, on: application.db.eventLoop)
-            .flatMapEach(on: application.db.eventLoop) { result -> EventLoopFuture<Void> in
-                switch result {
-                    case .success:
-                        pkg.status = .ok
-                    case .failure(let error):
-                        application.logger.error("Analysis error: \(error.localizedDescription)")
-                        pkg.status = .analysisFailed
-                }
-                return pkg.save(on: application.db)
+        if updates.isEmpty {
+            // Make sure we mark pkg as updated even if it has no versions - we don't want to
+            // get stuck reprocessing it all the time
+            // Simply mark it updated, don't touch the status
+            return pkg.update(on: application.db).map { [] }
+        } else {
+            return EventLoopFuture.whenAllComplete(updates, on: application.db.eventLoop)
+                .flatMapEach(on: application.db.eventLoop) { result -> EventLoopFuture<Void> in
+                    switch result {
+                        case .success:
+                            pkg.status = .ok
+                        case .failure(let error):
+                            application.logger.error("Analysis error: \(error.localizedDescription)")
+                            pkg.status = .analysisFailed
+                    }
+                    return pkg.save(on: application.db)
+            }
```

Comment:
This whole `analyze` function needs some refactoring and cleanup but I don’t want to do that before we’ve fixed the `wait` workaround.